core/server/middleware/request-id.js

```diff
@@ -0,0 +1,13 @@
+var uuid = require('uuid'),
+    _ = require('lodash'),
+    headerName = 'X-Request-Id',
+    idAttribute = 'requestId';
+
+module.exports = function requestId(req, res, next) {
+    req.userId = _.get(req, 'user.id', null);
```

Comment:
It's used for logging, see https://github.com/TryGhost/Ignition/blob/master/lib/logging/GhostLogger.js#L211.
This is was Ignition expects. You can see the output if you start Ghost with `MODE=long` or we could see the `userId` in any external logging tool later e.g. in loggly.